lib/ip_record.js

```diff
@@ -3,37 +3,96 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 // Keep track of events related to just IP addresses
-module.exports = function (BLOCK_INTERVAL_MS, now) {
+module.exports = function (BLOCK_INTERVAL_MS, RATE_LIMIT_INTERVAL_MS, MAX_ACCOUNT_STATUS_CHECK, now) {
 
   now = now || Date.now
 
+  var ACCOUNT_STATUS_ACTION = {
+    requestAccountStatus       : true
+  }
+
   function IpRecord() {}
 
+  function isAccountStatusAction(action) {
+    return ACCOUNT_STATUS_ACTION[action]
+  }
+
   IpRecord.parse = function (object) {
     var rec = new IpRecord()
     object = object || {}
     rec.bk = object.bk // timestamp when the account was blocked
+    rec.as = object.as || []       // timestamps when account status check occurred
+    rec.rl = object.rl       // timestamp when the account was rate-limited
```

Comment:
nit: comment indentation is inconsistent with the preceeding line; I don't have any particular opinion of the *correct* amount of indentation, but I do care about consistency :-)